### DISCUSSION TITLE
s3 integration programatically?

### DISCUSSION THREAD

**Discussion by tooptoop4** [drakkan/sftpgo#848] [General]
https://github.com/drakkan/sftpgo/blob/f2b93c04020f399e4901499ad2b3804773027502/docs/howto/postgresql-s3.md#add-virtual-users mentions clicking through the UI to configure sftp to s3. can it be done programmatically from config/cli/api call?

**Comment by drakkan** [comment]
Yes, you can use [REST API](https://sftpgo.stoplight.io/docs/sftpgo/68a341c676710-sftp-go)

**Reply by tooptoop4** [reply]
it seems no need, if my external auth returns the s3 details in the json

**Reply by drakkan** [reply]
> it seems no need, if my external auth returns the s3 details in the json

yes, external auth or pre-login hooks are another way
